File: week-3/is-subsequence/is-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
    if (!s) return true;
    
    let index = 0;
    for (let i of t) {
        if (i === s[index])
            ++index;
        if (index === s.length) return true;
    }
    
    return false;
};

File: week-3/is-subsequence/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if not s: return True
        
        index = 0
        for i in t:
            if i == s[index]:
                index += 1
            if index == len(s):
                return True
                
        return False
